### PR TITLE
[PR] Upgrading to `Phoenix 1.7`

### LINKED ISSUE #40: In case people can't determine if Postgresql is running with lsof -i :5432
Hi there,

This is my first ever open-source contribution post on GitHub.  I'm super-new at this but also want to make sure others don't get stuck with frustrating issues when learning.

I noticed that when going through the tutorial, there was a recommendation to use `lsof -i :5432` to check if Postgresql was running correctly.  This didn't work for me (there was no output), but I do know that Postgres was running as I had just successfully created a new DB with Ecto.

**Instead, I used `pg_isready` and got the desired confirmation:**

image

For context, I am running Ubuntu 18.04 through WSL2 with Postgresql-12.

Cheers,
Leo.

### LINKED ISSUE #149: Chore: Comprehensively Update this Tutorial to `Phoenix v1.7`
With the recent release of [`Phoenix v1.7.0-rc.2`](https://github.com/phoenixframework/phoenix/releases/tag/v1.7.0-rc.2) (which looks reasonably stable),
Now is a good time to _comprehensively_ update this tutorial to the latest and greatest.
This is a great opportunity for us to see the differences between `1.6.X` and `1.7` in a more involved project.

# Todo

+ [x] `git clone` this project to your `localhost` and rename it to `phx-chat-OLD`
+ [x] use this "old" version as the basis for creating the new one, but ...
+ [x] `git clone` the project _again_ and then ***`DELETE`*** all the code in the directories and `mix.exs`/`mix.lock` file:

phoenix-chat-example-delete

+ [x] Create a new branch on your `localhost` e.g. `phx-1.7-issue#X` and 
+ [x] Follow the instructions in the `REAMDE.md` to create a new app from scratch.
+ [x] Please remember to exclude all the noise that we don't need in a demo app like this e.g:

```sh
mix phx.new chat --no-mailer --no-dashboard --no-gettext
```

We _really_ don't need all the noise from constant @dependabot updates for `deps` we aren't using in the example! ⏳ 

> **Note**: this tutorial will ***NOT*** use `LiveView`. 
We have a different one for that: https://github.com/dwyl/phoenix-liveview-chat-example
That is deliberate. We still use "MVC" (even though `Phoenix` has removed the "V" ... https://www.germanvelasco.com/blog/phoenix-1-7-is-view-less ... 🙄 )
So we want `this` tutorial to continue to showcase the `M🙃C`   workflow even if we use `LiveView` in some other places.
I have _zero_ intention of using `LiveView` for the basic flow in `Auth` for example. It's just not appropriate. 

+ [x] Update the `README.md` as needed to match the updates in `Phoenix`.
+ [x] Match all the features in the current example including the `Tailwind` UI/UX and `auth` 
  + [x] update `auth.md` as needed (shouldn't be too much)
+ [x] Create a Pull Request with the update.
+ [x] Review your own PR ***`before`*** assigning it for review.
+ [x] Assign the PR to @SimonLab for review. 🙏 

I expect this update to take a few hours. (**`T4h`**? 🤷‍♂️)
It _pains_ me that we have to spend this time. I _wish_ there weren't so many **_breaking_ changes**! 😢 
But the _faster_ we can make this update the faster I (@nelsonic) can learn what I need to re-build `auth` https://github.com/dwyl/auth/issues/207 🤞 

Thanks! 🙏

### PR DESCRIPTION
**Pull request by LuchoTurtle**
closes #149 
closes #40 

This PR updates the project files to using `Phoenix 1.7-rc2` and the `README` and `auth.md` files.

### CODE REVIEW

**Comment by LuchoTurtle** [comment]
This upgrade should now be completed.
I've reviewed each change, the README and the `auth.md` files.

The app should run normally as before.

The `README` had a few outdated sections that were modified/deleted appropriately to reflect `Phoenix 1.7`.

**Comment by LuchoTurtle** [comment]
@SimonLab all yours. Should be a fairly simple review. If you have any trouble running it on your computer, give me a heads up.

**Comment by LuchoTurtle** [comment]
To install Phoenix 1.7, https://www.phoenixframework.org/blog/phoenix-1.7-released

**Comment by nelsonic** [comment]
Keen to pick this up soon. @SimonLab please assign to me once you've had a look. 🙏

File: .formatter.exs [hunk 1]
```diff
@@ -1,5 +0,0 @@
-[
-  import_deps: [:ecto, :phoenix],
-  inputs: ["*.{ex,exs}", "priv/*/seeds.exs", "{config,lib,test}/**/*.{ex,exs}"],
-  subdirectories: ["priv/*/migrations"]
-]
```
Comment (SimonLab) [inline .formatter.exs:5]: I think this file was delete by error. I've just checked by creating a new Phoenix 1.7 project and the file is still created automatically. I'll add it back and run `mix format`

File: lib/chat_web/channels/room_channel.ex [hunk 2]
```diff
@@ -2,13 +2,13 @@ defmodule ChatWeb.RoomChannel do
   use ChatWeb, :channel
 
   @impl true
-  def join("room:lobby", _payload, socket) do
-    #if authorized?(payload) do
+  def join("room:lobby", payload, socket) do
+    if authorized?(payload) do
```
Comment (SimonLab) [inline lib/chat_web/channels/room_channel.ex:6]: `authorized?` always returns `true` (see code below).

File: lib/chat_web/channels/room_channel.ex [hunk 3]
```diff
@@ -22,26 +22,30 @@ defmodule ChatWeb.RoomChannel do
...
+    end)
+
+    # :noreply
+    {:noreply, socket}
+  end
+
+  # Add authorization logic here as required.
+  defp authorized?(_payload) do
+    true
   end
```
Comment (SimonLab) [inline lib/chat_web/channels/room_channel.ex:50]: Is this function just a placeholder for future code logic? Or just an example how we can add authorization code?

Comment (LuchoTurtle) [reply lib/chat_web/channels/room_channel.ex:50]: This is boilerplate code generated by Phoenix when running `mix phx.gen.channel Room`. I didn't change it as it wasn't relevant to the tutorial 👍

File: mix.exs [hunk 4]
```diff
@@ -86,14 +70,12 @@ defmodule Chat.MixProject do
   # See the documentation for `Mix` for more info on aliases.
   defp aliases do
     [
-      "assets.deploy": ["tailwind default --minify", "esbuild default --minify", "phx.digest"],
-      c: ["coveralls.html"],
```
Comment (SimonLab) [inline mix.exs:90]: I think we can keep this command

Comment (LuchoTurtle) [reply mix.exs:90]: `mix c`?  I agree. I created the project from scratch using Phoenix 1.7 generator, so custom commands were defo overridden. 👍

**Review by SimonLab — APPROVED** [review]
Working on my machine and looks good to me :+1: 
@nelsonic I'll let you double check

File: .formatter.exs (hunk 1 above)
Comment (nelsonic) [reply .formatter.exs]: good spot. thanks for restoring. 👌

File: lib/chat_web/channels/room_channel.ex (hunk 2 above)
Comment (nelsonic) [reply lib/chat_web/channels/room_channel.ex:6]: this `authorized?/1` is a placeholder. 👌

File: mix.exs (hunk 4 above)
Comment (nelsonic) [reply mix.exs:90]: Restored _manually_ on the branch. 

```sh
n@mbp16 phoenix-chat-example % mix c
............
Finished in 0.5 seconds (0.5s async, 0.04s sync)
12 tests, 0 failures

Randomized with seed 414798
----------------
COV    FILE                                        LINES RELEVANT   MISSED
100.0% lib/chat.ex                                     9        0        0
100.0% lib/chat/message.ex                            26        4        0
100.0% lib/chat/repo.ex                                5        0        0
100.0% lib/chat_web/channels/room_channel.ex          51       11        0
100.0% lib/chat_web/components/layouts.ex              5        0        0
100.0% lib/chat_web/controllers/auth_controller       14        2        0
100.0% lib/chat_web/controllers/error_html.ex         19        1        0
100.0% lib/chat_web/controllers/error_json.ex         15        1        0
100.0% lib/chat_web/controllers/page_controller        9        1        0
100.0% lib/chat_web/controllers/page_html.ex           9        1        0
100.0% lib/chat_web/endpoint.ex                       49        0        0
100.0% lib/chat_web/router.ex                         32        5        0
[TOTAL] 100.0%
----------------
Generating report...
```

**Comment by nelsonic** [comment]
Running this on `localhost` I get the following `404` error in my terminal when accessing the page:
```sh
[info] GET /favicon.ico
[debug] ** (Phoenix.Router.NoRouteError) no route found for GET /favicon.ico (ChatWeb.Router)
    (chat 0.1.0) lib/phoenix/router.ex:482: ChatWeb.Router.call/2
    (chat 0.1.0) lib/chat_web/endpoint.ex:1: ChatWeb.Endpoint.plug_builder_call/2
    (chat 0.1.0) lib/plug/debugger.ex:136: ChatWeb.Endpoint."call (overridable 3)"/2
    (chat 0.1.0) lib/chat_web/endpoint.ex:1: ChatWeb.Endpoint.call/2
    (phoenix 1.7.0-rc.2) lib/phoenix/endpoint/sync_code_reload_plug.ex:22: Phoenix.Endpoint.SyncCodeReloadPlug.do_call/4
    (plug_cowboy 2.6.0) lib/plug/cowboy/handler.ex:11: Plug.Cowboy.Handler.init/2
    (cowboy 2.9.0) /Users/n/code/phoenix-chat-example/deps/cowboy/src/cowboy_handler.erl:37: :cowboy_handler.execute/2
    (cowboy 2.9.0) /Users/n/code/phoenix-chat-example/deps/cowboy/src/cowboy_stream_h.erl:306: :cowboy_stream_h.execute/3
    (cowboy 2.9.0) /Users/n/code/phoenix-chat-example/deps/cowboy/src/cowboy_stream_h.erl:295: :cowboy_stream_h.request_process/3
    (stdlib 4.2) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
```
Not a big deal, but we definitely don't want to ship this with errors ... 💭

**Review by nelsonic — APPROVED** [review]
@LuchoTurtle looks great! 👌 
@SimonLab thanks for first review. 🙏 
Ship. :shipit:

**Pull request merged**